### DISCUSSION TITLE
traffic seplit help

### DISCUSSION THREAD

**Discussion by 3157696** [strongswan/strongswan#2037] [Q&A]
HELLO
Currently our branches use openwrt(21.07) to connect to the gateway of the head office
It is easy to send traffic to the head office using the following command：
iptables -t nat -A POSTROUTING -s 10.0.3.0/24 -o eth0 -m policy --dir out --pol ipsec -j ACCEPT
https://docs.strongswan.org/docs/5.9/howtos/forwarding.html （1）

Now we plan to upgrade the branch's openwrt to 23.05,but it uses nft instead of iptables
However, neither the above nor the following methods met expectations. （1）and（2）
https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html#_marks_on_linux （2）
`nft insert rule inet fw4 mangle_prerouting ip saddr x.x.x.x ct mark set 0xa`
or
`nft insert rule inet fw4 forward meta ipsec exists  ip saddr x.x.x.x counter accept`

How to make specific traffic go through vpn like openwrt-21.07?

BTW
postrouting does not support meta ipsec exists？Are -m policy --dir out --pol ipsec equivalent to meta ipsec exists?
Thanks.

**Comment by tobiasbrunner** [comment]
Did you try `ipsec out` instead of `meta ipsec exists` (see `IPSEC EXPRESSIONS` in the [man page](https://www.netfilter.org/projects/nftables/manpage.html))? Not sure if it's still possible to use this without reqid, SPI or one of the addresses (source or destination), or whether wildcards are possible. The `iptables-nft` tool can translate the old syntax, but I think it does so directly via Netlink, maybe `nft` does not support this.
